lib/editor/components/pattern/TripPatternViewer.js

```diff
@@ -37,7 +37,7 @@ export default class TripPatternViewer extends Component<Props> {
     if (!activePattern) return null
 
     activePattern.patternLocations = activePattern.patternLocations.map(pl => { return {...pl, id: `${pl.locationId}${pl.stopSequence}`} })
-    activePattern.patternLocationGroups = activePattern.patternLocationGroups.map(plg => { return {...plg, id: `${plg.locationGroupId}${plg.stopSequence}`} })
+    activePattern.patternStopAreas = activePattern.patternStopAreas.map(psa => { return {...psa, id: `${psa.areaId}${psa.stopSequence}`} })
     activePattern.patternStops = activePattern.patternStops.map(ps => { return {...ps, id: `${ps.stopId}${ps.stopSequence}`} })
     return (
       <div>
```

Comment:
One side effect of this change: how the pattern name ("4 stops from A to B") is generated needs to be updated to use area names. When creating a pattern between areas, that text is now "4 stops from null to null".

Reply:
This is actually a historical issue that comes from the backend not checking if a name for a locationGroup (now stop Area) has not been defined. I have added a comment on the backend PR for this.